### ISSUE TITLE
Rule Change: rule [require-await]  consider async iterator function

### ISSUE THREAD

**Issue by ziloen** [eslint/eslint#17790]
### What rule do you want to change?

require-await consider async iterator `yield` a promise without `await` keyword

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
async function *asyncIterator() {
  let i = 0
  while(true) {
    i++;
    yield new Promise(resolve => setTimeout(() => resolve(i), 1000));
  }
}
```

### What does the rule currently do for this code?

Warning message: Async generator function 'asyncIterator' has no 'await' expression.

### What will the rule do after it's changed?

1. If `yield` a Promsie, then no warning.
2. Or add an option to ignore async iterator function.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

**Comment by mdjermanovic** [comment]
Hi @ziloen, thanks for the issue! 

> ### Example code
> ```js
> async function *asyncIterator() {
>   let i = 0
>   while(true) {
>     i++;
>     yield new Promise(resolve => setTimeout(() => resolve(i), 1000));
>   }
> }
> ```
> 
> ### What does the rule currently do for this code?
> Warning message: Async generator function 'asyncIterator' has no 'await' expression.

Which version of ESLint are you using? As of v7.0.0, the `require-await` rule ignores async generators. Here's a [Playground example](https://eslint.org/play/#eyJ0ZXh0IjoiLyplc2xpbnQgcmVxdWlyZS1hd2FpdDogXCJlcnJvclwiKi9cblxuLy8gbm8gZXJyb3JzXG5hc3luYyBmdW5jdGlvbiAqYXN5bmNJdGVyYXRvcigpIHtcbiAgbGV0IGkgPSAwXG4gIHdoaWxlKHRydWUpIHtcbiAgICBpKys7XG4gICAgeWllbGQgbmV3IFByb21pc2UocmVzb2x2ZSA9PiBzZXRUaW1lb3V0KCgpID0+IHJlc29sdmUoaSksIDEwMDApKTtcbiAgfVxufSIsIm9wdGlvbnMiOnsiZW52Ijp7fSwicnVsZXMiOnt9LCJwYXJzZXJPcHRpb25zIjp7ImVjbWFGZWF0dXJlcyI6e30sImVjbWFWZXJzaW9uIjoibGF0ZXN0Iiwic291cmNlVHlwZSI6Im1vZHVsZSJ9fX0=) with the latest version of ESLint.

**Comment by ziloen** [comment]
ESLint 8.54.0
@typescript-eslint/eslint-plugin 6.12.0
I didn't realize it was [@typescript-eslint/require-await](https://typescript-eslint.io/rules/require-await), I'll open an issue there.
Thank you!❤

**Comment by mdjermanovic** [comment]
Thanks for the info, closing as this issue isn't related to the core rule.
